### DISCUSSION TITLE
Some questions on the fire extinguishers that need a bit of research on.

### DISCUSSION THREAD

**Discussion by ageekhere** [CorsixTH/CorsixTH#2405] [Research Room]
I have a few questions on fire extinguishers that need a bit of research on (maybe even from looking at the source code) to find the correct answer.

1. Do fire extinguishers have any affect in rooms with no machines like the GP's office, if they do affect staff happiness does adding more than 1 change anything? 
2. Does adding more than 1 fire extinguisher in rooms with machines affect breakdowns?
3. Does position matter? if you have a very big room does having the fire extinguisher next to the machine affect anything?  
There may be a math formula for the fire extinguishers that could answer these questions.

**Comment by lewri** [comment]
From a CorsixTH perspective:

> * Do fire extinguishers have any affect in rooms with no machines like the GP's office, if they do affect staff happiness does adding more than 1 change anything?

Yes, they affect happiness a small amount, but only one extinguisher has an effect within a radius of two tiles.
> * Does adding more than 1 fire extinguisher in rooms with machines affect breakdowns?

It affects risk of room explosions, slightly.
> * Does position matter? if you have a very big room does having the fire extinguisher next to the machine affect anything?
>   There may be a math formula for the fire extinguishers that could answer these questions.

As above. An extinguisher placement will not change its effect on a machine exploding in a room. However, the radius to a staff member can matter.

**Reply by MandrakeHorse** [reply]
Do fire extinguishers improve the opinion of VIPs? If so, is there a direct effect (i.e. the VIP inspects each room and will be slightly impressed if it contains an extinguisher) or is it indirect (i.e. fire extinguishers make staff happy, which makes the VIP happy)?

**Reply by lewri** [reply]
I rewrote code for both the VIP and machines a few years ago. Now I feel sad I don't remember it. But yes a fire extinguisher affects the rating of a room for a VIP. Only one needs to be present.

See https://github.com/CorsixTH/CorsixTH/blob/87858c2d5a4060a6039ca4110fb24403d17d4123/CorsixTH/Lua/entities/humanoids/vip.lua#L183

**Reply by ageekhere** [reply]
Ah, did not know about the 2 tile fire extinguisher radius.
Which 2 tile radius is correct, the one on the left or right?
![Screenshot 2023-09-24 093035](https://github.com/CorsixTH/CorsixTH/assets/13173133/1a56439a-db4f-4cab-b7ae-3a45ea152ac7)

**Comment by ARGAMX** [comment]
Related to risk of room explosions only ~~first 3~~ first 4 extinguishers matter. 
Fourth and all subsequent ones is not counts.

https://github.com/CorsixTH/CorsixTH/blob/87858c2d5a4060a6039ca4110fb24403d17d4123/CorsixTH/Lua/entities/machine.lua#L123

https://github.com/CorsixTH/CorsixTH/blob/87858c2d5a4060a6039ca4110fb24403d17d4123/CorsixTH/Lua/entities/machine.lua#L113-L118

**Reply by ageekhere** [reply]
This is good research and now makes it clear on the num of extinguishers to use. Though I wonder if using diminishing returns logic would work better rather then using a hard cap.

**Reply by lewri** [reply]
Whoops, thanks for also showing here the comment and the code itself doesn't match (comment says 4 extinguishers, code says 3)

**Reply by ARGAMX** [reply]
> Whoops, thanks for also showing here the comment and the code itself doesn't match (comment says 4 extinguishers, code says 3)

Whoops, actually my bad. 😁

I have rechecked code on the fresh head and found that
the cycle will give 4, not 3, as max value for num_extinguishers. 🙂

So code and comment is consistent, but bit hard to compile in mind 😂

So actually Related to risk of room explosions only first 4 extinguishers matter. 🙂
